### DISCUSSION TITLE
Add a watch time/chat time in Viewer List

### DISCUSSION THREAD

**Discussion by tintinn** [Chatterino/chatterino2#3671] [Ideas] [answered]
This might be out of the scope for this project, but it'd be an interesting tool to have.

I would say that stream sniping is pretty common, and a lot of users have the same name as their IGN. Having a current view time/chat time since they joined the viewer list would help identify bad actors

**Comment by Felanbird** [comment, marked as answer]
There's 2 options for this request:
1. Add a system where Chatterino keeps track of time connected to chat.
2. Query the existing Streamelements "watchtime" stats. 

Both of which sound like they'd be extremely unreasonable to complete, I for one wouldn't want my Chatterino keeping track of every single users watchtime, I couldn't even imagine how much strain that would put on the application.

As for the Streamelements "watchtime", the same argument against it applies, having to query EVERY user in the viewer-list for their watchtime sounds awful (if watchtime even has an api call).

**Reply by tintinn** [reply]
Fair enough

I was thinking it could just record the current time when the user connected and show the difference between the current time on  request. 

Thanks for the response, and feel free to close this discussion

**Reply by LosFarmosCTL** [reply]
It might be somewhat doable by querying the user list once on connection, saving all users with a current timestamp and then using the membership capability to update that list on JOIN/PARTs. I have never used that though and from what I have heard it is apparently very flaky and not reliable.

Also, the twitch website dis-/reconnects you when you minimize/expand the chat, so that would probably mess with that watchtime as well.
Would need quite some testing whether it is in any way reliable.

**Reply by Felanbird** [reply]
>  feel free to close this discussion

Yeah if only discussions could be closed. 🙃
